pkg/multicluster/o11n.go

```diff
@@ -98,18 +99,28 @@ func GetClusterInfo(_ctx context.Context, k8sClient client.Client, clusterName s
 }
 
 // GetClusterMetricsFromMetricsAPI retrieves current cluster metrics based on GetNodeMetricsFromMetricsAPI
-func GetClusterMetricsFromMetricsAPI(ctx context.Context, k8sClient client.Client, clusterName string) (*ClusterUsageMetrics, error) {
+func GetClusterMetricsFromMetricsAPI(_ctx context.Context, k8sClient client.Client, clusterName string) (*ClusterUsageMetrics, error) {
 	nodeMetricsList := metricsV1beta1api.NodeMetricsList{}
-	if err := k8sClient.List(ContextWithClusterName(ctx, clusterName), &nodeMetricsList); err != nil {
+	ctx := ContextWithClusterName(_ctx, clusterName)
+	if err := k8sClient.List(ctx, &nodeMetricsList); err != nil {
 		return nil, errors.Wrapf(err, "failed to list node metrics")
 	}
-	var memoryUsage, cpuUsage resource.Quantity
+	var memoryUsage, cpuUsage, podUsage resource.Quantity
 	for _, nm := range nodeMetricsList.Items {
 		cpuUsage.Add(*nm.Usage.Cpu())
 		memoryUsage.Add(*nm.Usage.Memory())
 	}
+
+	// get the number of pods in cluster
+	podList := corev1.PodList{}
+	if err := k8sClient.List(ctx, &podList); err != nil {
```

Comment:
The list operation of pod could be heavy. For example, in one cluster, there might be tens of thousands of pods inside. The list operation can be time-consuming. Let's hold back the podUsage for now and discuss it more.